pkg/scheduler/scheduler.go

```diff
@@ -499,10 +494,15 @@ func (sched *Scheduler) scheduleOne() {
 		// into the resources that were preempted, but this is harmless.
 		if fitError, ok := err.(*core.FitError); ok {
 			preemptionStartTime := time.Now()
-			sched.preempt(pod, fitError)
-			metrics.PreemptionAttempts.Inc()
-			metrics.SchedulingAlgorithmPremptionEvaluationDuration.Observe(metrics.SinceInMicroseconds(preemptionStartTime))
-			metrics.SchedulingLatency.WithLabelValues(metrics.PreemptionEvaluation).Observe(metrics.SinceInSeconds(preemptionStartTime))
+			if !util.PodPriorityEnabled() || sched.config.DisablePreemption {
```

Comment:
What is the benefit of moving the check here?
I would prefer the logic of this function be simpler and easier to read and give the responsibility of the checking whether preemption is enabled or disabled to the "preempt" function.

Reply:
but when preemption is disabled, preemption did not occur, at this time I think metrics data shouldn't be count.

Reply:
I see. You are right. Please move `preemptionStartTime := time.Now()` from line 496 to the `else` block.

Reply:
I have moved it, thanks.